res/controllers/Denon-MC6000MK2-scripts.js

```diff
@@ -767,14 +767,16 @@ DenonMC6000MK2.OldDeck.prototype.hasLoopEnd = function () {
 };
 
 DenonMC6000MK2.OldDeck.prototype.hasLoop = function () {
-    return this.hasLoopStart() && this.hasLoopEnd();
+    return this.hasLoopStart() && this.hasLoopEnd() && this.getValue("loop_start_position") < this.getValue("loop_end_position");
 };
 
 DenonMC6000MK2.OldDeck.prototype.deleteLoopStart = function () {
+    this.setValue("loop_in", false);
     this.setValue("loop_start_position", DenonMC6000MK2.MIXXX_LOOP_POSITION_UNDEFINED);
 };
 
 DenonMC6000MK2.OldDeck.prototype.deleteLoopEnd = function () {
+    this.setValue("loop_out", false);
```

Comment:
FYI, with `component.Button`, resetting `loop_out` to 0 when pressing shift would be done automatically. https://mixxx.org/wiki/doku.php/components_js#shift_layers

Reply:
I did the switch ending up with a mixture of new and legacy code. Migration to components.js should be done all at once for each functional section. Maybe next (major) release ;)